src/components/call/ContactOffices.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { TranslationFunction } from 'i18next';
+import { translate } from 'react-i18next';
+import { Issue, Contact, DefaultContact } from '../../common/model';
+
+interface Props {
+  readonly currentIssue: Issue;
+  readonly contactIndex: number;
+  readonly t: TranslationFunction;
+}
+
+export interface State {
+  showFieldOfficeNumbers: boolean;  
+}
+
+export class ContactOffices extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {showFieldOfficeNumbers: false};
+  }
+
+  showField = () => {
+    this.setState({showFieldOfficeNumbers: true});
+  }
+
+  // this component is reused and the local state is maintained through contact changes,
+  // we want the local state to reset when it's updated
+  componentWillReceiveProps(nextProps: Props) {
+    this.setState({showFieldOfficeNumbers: false});    
```

Comment:
You can check here if the specific newProps that you are watching are different than the existing props(this.props).
https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops

> componentWillReceiveProps() is invoked before a mounted component receives new props. If you need to update the state in response to prop changes (for example, to reset it), you may compare this.props and nextProps and perform state transitions using this.setState() in this method.

> Note that React may call this method even if the props have not changed, so make sure to compare the current and next values if you only want to handle changes. This may occur when the parent component causes your component to re-render.